ddtrace/internal/_encoding.pyx

```diff
@@ -152,14 +154,16 @@ cdef class Packer(object):
 
         raise TypeError("Unhandled text type: %r" % type(text))
 
-    cdef inline int _pack_meta(self, object meta):
+    cdef inline int _pack_meta(self, object meta, object dd_origin):
```

Comment:
I don't see what we're gaining by special-casing `dd_origin` here. It follows the exact same code-path as any other key-value pair in the meta dict. I worry a bit about the precedent that this sets in the code for other trace-level tags.

Have we considered the approach where the origin tag is set into the meta dictionary right before this method? I would think that this approach would not result in a significant change to performance or would at least match the overhead of additional arguments/None checks that need to be done here.

Reply:
That's a fair point, and definitely simplifies the changes we'd need to make to the Packer. The reason I special cased `dd_origin` here may be a micro-optimization, but was done to avoid any unnecessary adding to the span meta. 

To be fair, this approach of using the encoder to add origin tagging isn't really benefiting by using the encoder at all, we're really only benefiting from using cython. From the benchmarks I've tested, there really isn't much of a performance benefit from using either method, and if the only difference between the two methods is essentially where `span.meta["_dd.origin"] = dd_origin` is implemented, I'd advocate for using the TraceTagProcessor (which is simpler and avoids messing around with the Packer) instead.

I do see that we can make further improvements to our encoder that would optimize dd origin tagging and other trace level tagging sending to the agent, but until we make those improvements (i.e. direct interaction with the buffer) at the moment I have to admit I'm leaning towards the TraceTagProcessor implementation.

@Kyle-Verhoog @brettlangdon WDYT?

Reply:
as long as performance numbers track to be about the same, I am ok with it. I think I am now more concerned with the overall performance add from this feature more so than individual implementations

Reply:
Next steps from offline discussion:

- Fix up the benchmark numbers for the different approaches in the PR description
- Add benchmarks for the current implementation against the proposed implementation above (add origin tag to meta in Cython)